Mathlib/CategoryTheory/Abelian/GrothendieckAxioms.lean

```diff
@@ -90,4 +95,32 @@ class AB5Star [HasCofilteredLimits C] where
 
 attribute [instance] AB5Star.preservesFiniteColimits
 
+noncomputable section
+
+open CoproductsFromFiniteFiltered
+
+variable {α : Type w}
+variable [HasZeroMorphisms C] [HasFiniteBiproducts C] [HasFiniteLimits C]
+
+instance preservesFiniteLimitsLiftToFinset : PreservesFiniteLimits (liftToFinset C α) :=
+  preservesFiniteLimitsOfEvaluation _ fun I =>
+    letI : PreservesFiniteLimits (colim (J := Discrete I) (C := C)) :=
+      preservesFiniteLimitsOfNatIso HasBiproductsOfShape.colimIsoLim.symm
+    letI : PreservesFiniteLimits ((whiskeringLeft (Discrete I) (Discrete α) C).obj
+        (Discrete.functor fun x ↦ ↑x)) :=
+      ⟨fun J _ _ => whiskeringLeftPreservesLimitsOfShape J _⟩
+    letI : PreservesFiniteLimits ((whiskeringLeft (Discrete I) (Discrete α) C).obj
+        (Discrete.functor (·.val)) ⋙ colim) :=
+      compPreservesFiniteLimits _ _
+    preservesFiniteLimitsOfNatIso (liftToFinsetEvaluationIso  I).symm
+
+/-- A category with finite biproducts and finite limits is AB4 if it is AB5. -/
+def AB4.ofAB5 [HasCoproducts C] [HasFilteredColimits C] [AB5 C] : AB4 C where
```

Comment:
A `HasFiniteCoproducts` assumption would be more consistent with the docstring, and it seems we have `hasCoproducts_of_finite_and_filtered`.